### DISCUSSION TITLE
The way spaces and rooms are organized makes no sense;

### DISCUSSION THREAD

**Discussion by expikr** [element-hq/element-meta#710] [Ideas]
#### What would you like to do?

clicking on a community should automatically list all of the rooms. The "Home" rooms list should simply serve as a shortcut folder for frequently accessed rooms, rather than being a strict container for loose rooms (and joining a space should not force-move the loose rooms to the space)

#### Why would you like to do it?

preserves object permanence

#### How would you like to achieve it?

clicking on a community should automatically list all of the rooms. The "Home" rooms list should simply serve as a shortcut folder for frequently accessed rooms, rather than being a strict container for loose rooms (and joining a space should not force-move the loose rooms to the space)

### Have you considered any alternatives?

Basically Discord, but with an extra "Frequent Channels" button.

![image](https://user-images.githubusercontent.com/77922942/194067009-7a6e01dc-6bf9-4b88-b709-f1cc417a7a77.png)

### Additional context

vector-im/element-web#23425

**Comment by t3chguy** [comment]
> clicking on a community should automatically list all of the rooms.

This is already possible if the Space marks its rooms as `Suggested`

**Comment by expikr** [comment]
> > clicking on a community should automatically list all of the rooms.
> 
> This is already possible if the Space marks its rooms as `Suggested`

Then the pattern should be reversed. Default-show and optional "mark unsuggested". This is a problem with the Matrix protocol itself, then?

**Comment by expikr** [comment]
_Originally posted by @DanMan in https://github.com/vector-im/element-web/issues/15930#issuecomment-863086706_:
> What's the reasoning behind not listing all the rooms inside a Space for everyone in that (private) Space? It seems like even a public room with guest access is not listed for all the Space members. Aren't they supposed to be able to discover all the (public) rooms? What's the point of "suggested" rooms then after all?

seems that it had been mentioned before but completely missed the point, the problem is with the reversed default being terrible UX, not a problem with private/public functionality.

**Reply by HarHarLinks** [reply]
This could perhaps be improved by a clientside implementation detail that will default rooms being created in (public?) spaces to suggested? Though it's not an entirely obvious decision that may have some downsides I haven't yet considered.

An alternative would be to improve the visibility/discoverability of the space homes, e.g. the last entry in the room list could be a button "this is the end of the room list, click to go to space home and explore more rooms in this space".

**Comment by t3chguy** [comment]
Moving to meta as this is a cross-platform concern

**Comment by expikr** [comment]
oh wtf there is infinite roomception? Technically impressive but terrible for user experience if not provided with an opinionated set of defaults/restrictions.

Discord's opinionated organization is:

* DMs
* Group Chats
* Servers
  - Categories (including uncategorized)
    - Channels
      - Threads

Matrix's, from what I could gather so far, is:
* DMs
* Rooms
* Spaces
  - Rooms
  - Subspaces
    - Rooms
    - Subspaces
      - Rooms
      - Subspaces
        - ...

**Reply by HarHarLinks** [reply]
Your understanding is basically correct.
Technically, in Matrix, DMs and Spaces are also Rooms, it is "only" client UI that differentiates them from each other UX-wise.
In the case of Element, you can configure to handle this in multiple ways to suit your needs. For example, DMs will _also_ be shown in Spaces when both of you are Space members, but you can turn it off per-space (right click a space) if you like.
Also, while a Room can appoint (a) "canonical" parent(s), it can be included in any and as many Spaces as anyone wants it to be. Particularly, you can create private spaces that e.g. sort and collects e.g. your favorite rooms from a couple spaces into your own private parallel hierarchy.
